tracing/tracing_test.go

```diff
@@ -0,0 +1,25 @@
+package tracing_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/birdie-ai/golibs/slog"
+	"github.com/birdie-ai/golibs/tracing"
+)
+
+func TestIntrumentedHTTPHandler(t *testing.T) {
+	var got *slog.Logger
+	handler := tracing.InstrumentHTTP(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		got = slog.FromCtx(req.Context())
+	}))
+
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	res := httptest.NewRecorder()
+	handler.ServeHTTP(res, req)
+
+	if got == nil {
+		t.Fatal("got nil logger")
+	}
```

Comment:
The tests are not great right now, I wanted to test if the trace_id field is there, test the header integration, etc. But baby steps =P.

Reply:
Yep, good points.